bundles/net.tourbook.cloud/src/net/tourbook/cloud/suunto/SuuntoCloudDownloader.java

```diff
@@ -125,130 +126,138 @@ private int downloadFiles(final List<Payload> newWorkouts, final IProgressMonito
+                           UI.SYMBOL_HOURGLASS_WITH_FLOWING_SAND }));
 
-            monitor.worked(1);
+               numberOfDownloadedTours[0] = downloadFiles(newWorkouts, monitor);
 
-            monitor.done();
+               monitor.worked(1);
 
-            return Status.OK_STATUS;
-         }
-      };
+               monitor.done();
+
+               return Status.OK_STATUS;
+            }
+         };
 
-      final long start = System.currentTimeMillis();
+         final long start = System.currentTimeMillis();
 
-      TourLogManager.log_TITLE(Messages.Log_DownloadWorkoutsFromSuunto_001_Start);
+         TourLogManager.log_TITLE(Messages.Log_DownloadWorkoutsFromSuunto_001_Start);
 
-      job.setPriority(Job.INTERACTIVE);
-      job.schedule();
+         job.setPriority(Job.INTERACTIVE);
+         job.schedule();
 
-      job.addJobChangeListener(new JobChangeAdapter() {
-         @Override
-         public void done(final IJobChangeEvent event) {
+         job.addJobChangeListener(new JobChangeAdapter() {
+            @Override
+            public void done(final IJobChangeEvent event) {
 
-            if (PlatformUI.isWorkbenchRunning()) {
+               if (PlatformUI.isWorkbenchRunning()) {
 
-               final String infoText = event.getResult().isOK()
-                     ? NLS.bind(Messages.Dialog_DownloadWorkoutsFromSuunto_Message,
-                           numberOfDownloadedTours[0],
-                           _numberOfAvailableTours[0] - numberOfDownloadedTours[0])
-                     : notificationText[0];
+                  final String infoText = event.getResult().isOK()
+                        ? NLS.bind(Messages.Dialog_DownloadWorkoutsFromSuunto_Message,
+                              numberOfDownloadedTours[0],
+                              _numberOfAvailableTours[0] - numberOfDownloadedTours[0])
+                        : notificationText[0];
 
-               PlatformUI.getWorkbench().getDisplay().asyncExec(() -> {
+                  PlatformUI.getWorkbench().getDisplay().asyncExec(() -> {
 
-                  TourLogManager.log_TITLE(String.format(LOG_CLOUDACTION_END, (System.currentTimeMillis() - start) / 1000.0));
+                     TourLogManager.log_TITLE(String.format(LOG_CLOUDACTION_END, (System.currentTimeMillis() - start) / 1000.0));
 
-                  UI.openNotificationPopup(
-                        Messages.Dialog_DownloadWorkoutsFromSuunto_Title,
-                        Activator.getImageDescriptor(CloudImages.Cloud_Suunto_Logo_Small),
-                        infoText);
-               });
+                     UI.openNotificationPopup(
+                           Messages.Dialog_DownloadWorkoutsFromSuunto_Title,
+                           Activator.getImageDescriptor(CloudImages.Cloud_Suunto_Logo_Small),
+                           infoText);
+                  });
+               }
             }
-         }
-      });
+         });
+      }
+      _isInDownload = false;
```

Comment:
I think that this flag do not work, this flag should be set to `false` at the end of  the running job and not when the job is scheduled, this returns immediately.

Reply:
`_isInDownload` may needs to be `static` it may also needs to be `final` then the trick is to use a `boolean[]` with one element

Reply:
Based on this, I have changed the fix to use a property of `_prefStore` instead of static. That way, if a new download is launched the current `_prefStore` is unchanged.

Reply:
I don't unterstand this new code, the reference of a pref store is static and I don't see a difference, when you apply the static pref store to a field, it returns always the same static pref store.

In `TourbookPlugin` I already simplified it

![image](https://github.com/mytourbook/mytourbook/assets/1283445/fdee356f-b866-4d24-82e5-d246ced6b864)

Reply:
I think the approach with `_isInDownload` was OK but it was reseted at the wrong time/location. When this would be a very critical concurrent case, then an AtomicBoolean could be used.

Reply:
That's a possibility but for the moment i think i prefer deactivating the person selector

Reply:
Actually, even using the boolean approach would not work because the issue here is that while the first download runs, if the preference store value changes, then the current download has erroneous results. Hence the fix to disable the person combo is the best so far.

>I don't unterstand this new code, the reference of a pref store is static and I don't see a difference, when you apply the static pref store to a field, it returns always the same static pref store.

My idea was to get a copy of the preference store that would be a copy, i.e.: it would not change when the user does actions during the download but evidently, it doesn't seem to be possible